scripts/check_regression.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# Check if two arguments are provided (paths to past.json and current.json)
+if [ $# -ne 2 ]; then
+    echo "Usage: $0 <path_to_past.json> <path_to_current.json>"
+    exit 1
+fi
+
+# Extract the file paths from command-line arguments
+past_json_path="$1"
+current_json_path="$2"
+
+# Read the contents of past.json and current.json into variables
+past_data=$(cat "$past_json_path")
+current_data=$(cat "$current_json_path")
+
+# Function to compare P90 values for a given statistic
+compare_stat_p90() {
+    local past_value="$1"
+    local current_value="$2"
+    local test_name="$3"
+    local stat_name="$4"
+
+    # Calculate 110% of the past value
+    local threshold=$(calculate_threshold "$past_value")
+
+    # Compare the current value with the threshold
+    if (( $(awk 'BEGIN {print ("'"$current_value"'" > "'"$threshold"'")}') )); then
+        echo "ERROR: $stat_name - Current P90 value ($current_value) exceeds the 110% threshold ($threshold) of the past P90 value ($past_value)"
+        return 1
+    fi
+
+    return 0
+}
+
+calculate_threshold() {
+    local past_value="$1"
+    awk -v past="$past_value" 'BEGIN { print past * 1.1 }'
+}
+
+# Loop through each object in past.json and compare P90 values with current.json for all statistics
+compare_p90_values() {
+    local past_json="$1"
+    local current_json="$2"
+
+    local test_names=$(echo "$past_json" | jq -r '.benchmarkTests[].testName')
+
+    for test_name in $test_names; do
+        echo "Testing - '$test_name'"
+        for stat_name in "fullRunStats" "pullStats" "lazyTaskStats" "localTaskStats"; do
+            local past_p90=$(echo "$past_json" | jq -r --arg test "$test_name" '.benchmarkTests[] | select(.testName == $test) | .'"$stat_name"'.pct90')
+            local current_p90=$(echo "$current_json" | jq -r --arg test "$test_name" '.benchmarkTests[] | select(.testName == $test) | .'"$stat_name"'.pct90')
+
+            compare_stat_p90 "$past_p90" "$current_p90" "$test_name" "$stat_name" || return 1
```

Comment:
Instead of returning here I think we should echo that we saw a regression keep but keep iterating through the rest of the image set/stats. This way we don't miss any other regressions.


We can note that we saw a regression with some var and return it at the end:

```
local regression = 0

if [ $? -eq 1 ]; then
$regression = 1
...


if [ $regression -eq 1 ]; then
    echo "Comparison failed. Regression detected."
    exit $regression
```

Reply:
Yup I think it's a good idea to identify all regressions at once. Updated